src/common/Finisher.cc

```diff
@@ -61,23 +58,27 @@ void *Finisher::finisher_thread_entry()
       ldout(cct, 10) << "finisher_thread doing " << ls << dendl;
 
       // Now actually process the contexts.
+      utime_t start;
       for (vector<Context*>::iterator p = ls.begin();
-	   p != ls.end();
-	   ++p) {
-	if (*p) {
-	  (*p)->complete(0);
-	} else {
-	  // When an item is NULL in the finisher_queue, it means
-	  // we should instead process an item from finisher_queue_rval,
-	  // which has a parameter for complete() other than zero.
-	  // This preserves the order while saving some storage.
-	  assert(!ls_rval.empty());
-	  Context *c = ls_rval.front().first;
-	  c->complete(ls_rval.front().second);
-	  ls_rval.pop_front();
-	}
-	if (logger) {
-	  logger->dec(l_finisher_queue_len);
+           p != ls.end();
+           ++p) {
+        if(logger)
+          start = ceph_clock_now(cct);
+        
+        if (*p) {
+          (*p)->complete(0);
+        } else {
+          // When an item is NULL in the finisher_queue, it means
+          // we should instead process an item from finisher_queue_rval,
+          // which has a parameter for complete() other than zero.
+          // This preserves the order while saving some storage.
+          assert(!ls_rval.empty());
+          Context *c = ls_rval.front().first;
+          c->complete(ls_rval.front().second);
+          ls_rval.pop_front();
+        }
+        if (logger) {
+          logger->dec(l_finisher_queue_len);
```

Comment:
minor: we can save the current time here and used to avoid set "start = ceph_clock_now(cct);" again in the loop.